cmd/antrea-controller/options.go

```diff
@@ -63,6 +66,39 @@ func (o *Options) validate(args []string) error {
 	if len(args) != 0 {
 		return errors.New("no positional arguments are supported")
 	}
+
+	if o.config.EnableNodeIPAMController {
+		// Validate ClusterCIDRs
+		cidrSplit := strings.Split(strings.TrimSpace(o.config.ClusterCIDRs), ",")
+		cidrs, err := netutils.ParseCIDRs(cidrSplit)
+
+		hasIP4, hasIP6 := false, false
+		for _, cidr := range cidrs {
+			if cidr.IP.To4() == nil {
+				hasIP6 = true
+			} else {
+				hasIP4 = true
+			}
+		}
+		if err != nil {
+			return fmt.Errorf("cluster CIDRs %s is invalid", o.config.ClusterCIDRs)
+		}
+		if hasIP4 && o.config.NodeCIDRMaskSizeIPv4 == 0 {
```

Comment:
Should not we check the value falls into the valid range (e.g. 8 - 31)?

Reply:
And if the default value is 24, then you should set it in setDefaults() when the parameter is not specified.

Reply:
I've removed the defaults as @salv-orlando requested. IDK if we need these defaults or not as K8S NodeIPAM does have default values for mask sizes.
I agree about range validation, either if we have defaults or not it's a good idea.

Reply:
I am fine to remove the defaults, but you should revise the config parameter description in the YAML (which still mentions defaults).